scene/3d/audio_stream_player_3d.cpp

```diff
@@ -595,83 +543,52 @@ void AudioStreamPlayer3D::seek(float p_seconds) {
 
 void AudioStreamPlayer3D::stop() {
 	setplay.set(-1);
-	for (Ref<AudioStreamPlayback> &playback : stream_playbacks) {
+	internal.stop();
+	for (Ref<AudioStreamPlayback> &playback : internal.stream_playbacks) {
 		AudioServer::get_singleton()->stop_playback_stream(playback);
 	}
-	stream_playbacks.clear();
-	active.clear();
-	set_physics_process_internal(false);
 }
 
 bool AudioStreamPlayer3D::is_playing() const {
-	for (const Ref<AudioStreamPlayback> &playback : stream_playbacks) {
-		if (AudioServer::get_singleton()->is_playback_active(playback)) {
-			return true;
-		}
-	}
 	if (setplay.get() >= 0) {
 		return true; // play() has been called this frame, but no playback exists just yet.
 	}
-	return false;
+	return internal.is_playing();
 }
 
 float AudioStreamPlayer3D::get_playback_position() {
-	// Return the playback position of the most recently started playback stream.
-	if (!stream_playbacks.is_empty()) {
-		return AudioServer::get_singleton()->get_playback_position(stream_playbacks[stream_playbacks.size() - 1]);
-	}
-	return 0;
+	return internal.get_playback_position();
 }
 
 void AudioStreamPlayer3D::set_bus(const StringName &p_bus) {
-	//if audio is active, must lock this
+	// If audio is active, must lock this.
 	AudioServer::get_singleton()->lock();
```

Comment:
The lock does not exist in other players. Should it be removed?

Reply:
Good catch. It should be unnecessary. We handle thread contention between mixing and modification of playback params like this in AudioServer with SafeList.